src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -345,7 +354,8 @@ abstract class ClassfileParser {
             if (index <= 0 || ConstantPool.this.len <= index) errorBadIndex(index)
             val start = firstExpecting(index, CONSTANT_UTF8)
             val len   = (in getChar start).toInt
-            in.buf drop start + 2 take len
+            val s     = start + 2
+            in.buf.slice(s, s + len)
```

Comment:
This could be `.view.slice` to avoid creating the intermediate arrays just to pass into `flatMap`.

Going further, we might be able to correctly size the result array by summing the lengths.

```scala
    def getBytes(indices: List[Int]): Array[Byte] = {
      val head = indices.head
      values(head) match {
        case xs: Array[Byte] => xs
        case _               =>
          val starts = new Array[Int](indices.length)
          val lengths = new Array[Int](indices.length)
          var totalLength = 0
          foreachWithIndex(indices) { (index, i) =>
            if (index <= 0 || ConstantPool.this.len <= index) errorBadIndex(index)
            val start = firstExpecting(index, CONSTANT_UTF8)
            starts(i) = start
            val length = (in getChar start).toInt
            lengths(i)  = length
            totalLength += length
          }
          val result = new Array[Byte](totalLength)
          var resultPos = 0
          for (i <- starts.indices) {
            System.arraycopy(in.buf, starts(i) + 2, result, resultPos, lengths(i))
            resultPos += lengths(i)
          }
          recordAtIndex(getSubArray(result), head)
      }
    }
```

But AFAIK, this isn't a hot path so long as we still store most pickles in a single string, rather than array, so perhaps this is a premature optimization.